### DISCUSSION TITLE
Updated dev container usage method.

### DISCUSSION THREAD

**Discussion by klutzzykitty** [blakeblackshear/frigate#5936] [General]
Wanted to know what's the correct way to start a development container with Version 0.12. 
I've used dev containers before with v0.10 and v0.10.1 and before it used to be 
```
Run python3 -m frigate to start the backend.
In a separate terminal window inside VS Code, change into the web directory and run npm install && npm start to start the frontend.
```
Noticed that after V0.11.0 release,  the start script is not present in [package.json](https://github.com/blakeblackshear/frigate/blob/dev/web/package.json)

before there used to be a start script with `snowpack` for V0.10, like so.  `"start": "cross-env SNOWPACK_PUBLIC_API_HOST=http://localhost:5000 snowpack dev",`

so, as the documentation still mentions running `npm start` , it results in start script not found error. 
Has the method to start the front end in a dev container changed ? 

Thanks : )

**Comment by NickM-27** [comment]
You were most likely using the stable docs along with the dev branch which is incorrect. In any case, 0.12 has been merged so the stable docs are now up to date https://docs.frigate.video/development/contributing

**Reply by NickM-27** [reply]
actually, maybe they're not. It should be `npm run dev` not `npm start`

**Reply by klutzzykitty** [reply]
@NickM-27  Yeah exactly. That's what I was thought. 
I even tried with `npm run dev` earlier before asking here. 

But looks like there's something missing. 
I've ran the python part i.e. `python3 -m frigate` , then running `npm run dev`, results in the below happening. 
although it doesn't output any error, but doesn't load the front-end either. 
No errors with the python modules. 

Usually I used to encounter these when Mqtt wasn't connected, Here however, Mqtt is connected. 

reference for `npm run dev`
![dev_cont_1](https://user-images.githubusercontent.com/118447577/230776492-58a736cc-5f9c-472d-912a-6d09a4bca150.png)

reference for what's seen on the front-end 
![dev_cont_2](https://user-images.githubusercontent.com/118447577/230776663-2faa9639-f6e2-4540-9d0e-3ac1851adcd5.png)

It just keeps loading. 
Any suggestions ? 
Thanks !!! : D

**Reply by NickM-27** [reply]
something doesn't seem right, it should be loading the frontend on :5173 not :3000 in the dev branch. 

https://github.com/blakeblackshear/frigate/issues/4957 may be helpful as well

**Reply by klutzzykitty** [reply]
Thanks for the quick reply ! 
I'll check out and see if it helps.

**Reply by klutzzykitty** [reply]
@NickM-27 
Manually running nginx executable  `exec ./nginx` within the dev-container solves the problem. 

Not sure why as other older versions didn't have an nginx issue in dev containers while following instructions. 

Thanks tho  :)
